pkgs/development/python-modules/uxsim/add-qt-plugin-path-to-env.patch

```diff
@@ -0,0 +1,24 @@
+diff --git a/uxsim/__init__.py b/uxsim/__init__.py
+index 01e1ad1..de1f0fd 100644
+--- a/uxsim/__init__.py
++++ b/uxsim/__init__.py
+@@ -1,8 +1,14 @@
+-from .uxsim import *
+-from .utils import *
++import os
++
+ from .analyzer import *
++from .utils import *
++from .uxsim import *
++
++# Only set our own plugin path if it's not already set
++if not os.getenv("QT_PLUGIN_PATH"): 
```

Comment:
this can be replaced by makeWrapper --set-defailt argument

https://nixos.org/manual/nixpkgs/unstable/#fun-makeWrapper

Reply:
on a second thought... that will not work for python module

Reply:
@kirillrdy i was very unsure of what the preferred method might be here since it is a module. i don't particularly like the way i've done it here but it does work. i would be happy if you or anyone else has a better suggestion :slightly_smiling_face:

Reply:
i don't think there is a better option.

you could suggest your patch upstream ?

Reply:
hmm does it really make sense to go upstream in this case? idk its kind of convoluted

the patch sets env `QT_PLUGIN_PATH` to `$NIX_QT_PLUGIN_PATH` which is just a dummy string that only exists to get replaced in in the derivation with `substituteInPlace` with the real path

the way the patch currently works, if upstreamed, would just set `QT_PLUGIN_PATH` to a dummy string which I suppose wouldn't break anything but also probably isn't something I'd want to add to my own codebase

idk :zany_face:

Reply:
yeah, i guess you are right